### PR TITLE
Added AppInstallResultEvent event

### PR DESCRIPTION
**Pull request by AmelBawa-msft**
## Summary of the pull request
-Added AppInstallResultEvent event

## References and relevant issues
https://task.ms/48744611

## PR checklist
- [ ] Closes #xxx
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

File: tools/SetupFlow/DevHome.SetupFlow/Models/InstallPackageTask.cs [hunk 1]
```diff
@@ -282,11 +282,11 @@ private void ReportAppSelectedForInstallEvent()
 
     private void ReportAppInstallSucceededEvent()
     {
-        TelemetryFactory.Get<ITelemetry>().Log("AppInstall_InstallSucceeded", LogLevel.Critical, new AppInstallEvent(_package.Id, _package.CatalogId), _activityId);
+        TelemetryFactory.Get<ITelemetry>().Log("AppInstall_InstallSucceeded", LogLevel.Critical, new AppInstallResultEvent(_package.Id, _package.CatalogId), _activityId);
```
Comment (dhoehna) [inline tools/SetupFlow/DevHome.SetupFlow/Models/InstallPackageTask.cs:285]: Please modify the name to use two `_'s` to keep in line with other event names.  Something like `AppPage_AppInstall_InstallSucceeded`

Comment (dhoehna) [inline tools/SetupFlow/DevHome.SetupFlow/Models/InstallPackageTask.cs:285]: For telemetry prefer to use one event name and to use a variable to separate the actions/result/operations.  For example the event name could be `AppPage_AppInstall_InstallEvent` and `AppInstallResultEvent` can have a `public bool DidSucceed` variable.

Comment (AmelBawa-msft) [reply tools/SetupFlow/DevHome.SetupFlow/Models/InstallPackageTask.cs:285]: The change in this PR address the telemetry PartA property. Changing the name may require further consideration. Check https://github.com/microsoft/devhome/pull/2809#discussion_r1591384308 for more details.

Comment (AmelBawa-msft) [reply tools/SetupFlow/DevHome.SetupFlow/Models/InstallPackageTask.cs:285]: Same comment as above

**Review by krschau — APPROVED** [review]

**Pull request merged**